src/PrestaShopBundle/ApiPlatform/Serializer.php

```diff
@@ -0,0 +1,122 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShopBundle\ApiPlatform;
+
+use ReflectionException;
+use ReflectionMethod;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+use Symfony\Component\Serializer\Serializer as SymfonySerializer;
+use Traversable;
+
+class Serializer implements NormalizerInterface, DenormalizerInterface
+{
+    private SymfonySerializer $serializer;
+
+    /**
+     * @param Traversable $normalizers
+     */
+    public function __construct(Traversable $normalizers)
+    {
+        $this->serializer = new SymfonySerializer(iterator_to_array($normalizers));
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws ReflectionException
+     */
+    public function denormalize($data, string $type, string $format = null, array $context = []): mixed
```

Comment:
That's some important logic... that I think is worth unit testing 😄 wdyt?

Reply:
For the moment integrations tests are used to check if all works fine

Reply:
resolved